Версия 5. <feat> Добавлен класс SqlQueryConstantsForFilm для хранения…

Solntmore opened the pull request:
… SQL запросов в БД для работы с фильмами.

          <feat> Добавлен класс SqlQueryConstantsForUser, для хранения SQL запросов в БД для работы с пользователями.
          <feat> Добавлены классы FilmDbStorage и UserDbStorage, выполняющие роль хранилища вместо InMemoryFilmStorage и InMemoryUserStorage.
          <feat> Добавлены классы Genre, Mpa, GenresController, MpaController, BaseEntity для работы с новыми сущностями.
          <refactor> Валидация перенесена на уровень Service.

Solntmore committed:
Версия 5. <feat> Добавлен класс SqlQueryConstantsForFilm для хранения SQL запросов в БД для работы с фильмами.
          <feat> Добавлен класс SqlQueryConstantsForUser, для хранения SQL запросов в БД для работы с пользователями.
          <feat> Добавлены классы FilmDbStorage и UserDbStorage, выполняющие роль хранилища вместо InMemoryFilmStorage и InMemoryUserStorage.
          <feat> Добавлены классы Genre, Mpa, GenresController, MpaController, BaseEntity для работы с новыми сущностями.
          <refactor> Валидация перенесена на уровень Service.

Fleaman777 commented on src/main/java/ru/yandex/practicum/filmorate/constants/SqlQueryConstantsForUser.java:
```diff
@@ -0,0 +1,24 @@
+            "WHERE first_user_id = :first_user_id AND second_user_id = :second_user_id";
+    public static final String SQL_QUERY_REMOVE_FROM_FRIENDS = "DELETE FROM friends WHERE first_user_id = :first_user_id " +
+            "AND second_user_id = :second_user_id";
+
```
неплохо придумано, вынесение всех запросов отдельный для этого класс

Fleaman777 commented on src/main/java/ru/yandex/practicum/filmorate/model/Genre.java:
```diff
@@ -0,0 +1,17 @@
+@AllArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+public class Genre extends BaseEntity {
+    String name;
```
поле должно быть private

Fleaman777 commented on src/main/java/ru/yandex/practicum/filmorate/storage/user/UserDbStorage.java:
```diff
@@ -0,0 +1,188 @@
+
+    public User createUser(User user) {
+        String name = user.getName();
+        if (name == null || name.isBlank()) {
```
для информации на будущее, 
есть очень хорошая библиотека StringUtils для работы со строками, от Apache, она подтягивается отдельно через депенденси в пом файле, скорее всего на работе будешь ее использовать,  

в целом, можешь уже сейчас начать ей пользоваться)
в ней есть метод isBlank(String) - который в себе содержит сразу вот эти 2 проверки name == null || name.isBlank()

Fleaman777 commented on src/test/java/ru/yandex/practicum/filmorate/FilmorateApplicationTests.java:
```diff
@@ -1,13 +1,256 @@
+    private final UserDbStorage userStorage;
+    private final FilmDbStorage filmStorage;
+    private final static LocalDate BIRTHDAY = LocalDate.of(1967, 03, 25);
+    private final static LocalDate TEST_DATE = LocalDate.of(1895, 12, 29);
```
по джава код стайлу, константы идут до обычных полей класса

Fleaman777 commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmDbStorage.java:
```diff
@@ -0,0 +1,216 @@
+                .query(SQL_QUERY_TAKE_ALL_GENRES, (rs, rowNum) -> makeGenre(rs));
+    }
+
+    public Optional<Genre> findGenreById(int genreId) {
```
для жанров и рейтингов лучше создать отдельные dbStorage классы,
тем самым мы больше разделим ответсвенность, + в будущем будет проще добавлять новый функционал, 
А так же его тестировать

Fleaman777 commented on src/test/java/ru/yandex/practicum/filmorate/FilmorateApplicationTests.java:
```diff
@@ -1,13 +1,256 @@
 @SpringBootTest
+@AutoConfigureTestDatabase
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
 class FilmorateApplicationTests {
```
обычно для ApplicationClass не создаются тесты, т.к. он никакой логики не содержит, только пару строк кода, которые запускают все приложение, 

а те тесты(тест на каждый публичный метод DbStorage), которые сейчас содержаться внутри этого класса, нужно разложить по соответствующим классам, например UserDbStorageTest и т.д.

Solntmore committed:
Версия 5. <feat> Добавлен классы GenreDbStorage, GenreService, MpaDbStorage, MpaService для разделения ответственности классов.
          <feat> Добавлен классы FilmDbStorageTest, GenreDbStorageTest, MpaDbStorageTest, UserDbStorageTest для тестирования публичных методов Storage.
          <refactor> В классе UserDbStorage добавлено использование библиотеки StringUtils.

Solntmore committed:
Версия 6.<refactor> Обновлено содержание файла readme.

Fleaman777 commented on README.md:
```diff
@@ -1,2 +1,34 @@
 # java-filmorate
 Template repository for Filmorate project.
+![Схема базы данных](src/main/resources/IMG_8495.jpg)
```
картинки не нашел, 
добавишь ее к следующему спринту

Solntmore committed:
Версия 7.<feat> Добавлена схема БД в папку resourses.
